tool/update_github_emojis.dart

```diff
@@ -0,0 +1,338 @@
+// Copyright (c) 2022, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:convert';
+import 'dart:io';
+
+import 'package:args/args.dart';
+
+import 'package:markdown/src/legacy_emojis.dart' as legacy;
+
+// REGEXP to parse github emoji api
+RegExp parseGithubEmojiApi =
+    RegExp(r'^[\ \t]+"(.*?)".*unicode\/([A-Fa-f0-9\-]+)\.png');
+
+RegExp parseGithubEmojiUnicodeFromFilename =
+    RegExp(r'.*unicode\/([A-Fa-f0-9\-]+)\.png');
+
+// Get our emoji list from Githubs api.  We reconcile with our legacy emojis so that
+// we don't change or break anything.
+// There are essentially only TWO (2) emojis that change and the
+// legacy emoji is still available with an alternate name.
+// The 'beetle' emoji changes from `🐞` to `🪲`, legacy available as 'lady_beetle'.
+// The 'cricket' emoji changes from `🏏` to `🦗`, legacy available as 'cricket_game'.
+// (if the -g flag us used to force using the Github unicode sequences for the
+// emojis then additionally the 'email' emoji changes from '✉️' to '📧')
+final _emojisJsonRawUrl = 'https://api.github.com/emojis';
+final _emojisFilePath = 'lib/src/emojis.dart';
+
+/// Reference to emojis map within legacy_emojis.dart
+final legacyEmojis = legacy.emojis;
+
+/// AUTO GENERATED by [reconcile_emojis.dart] - this only needed to be done ONCE
+/// during the reconciliation process with the legacy emojis.
+/// This array is ONLY USED when the --useGithubUnicodes option is used to
+/// minimize the visual differences in the output emojis.
+const legacyEmojisUsedVariationModifier = [
```

Comment:
this should/could be a `Set`?

Reply:
By default the currently set behavior in the code is to always use the exact same emoji sequences as we currently do (these become the 'legacy' emoji sequences in this PR)

The `const legacyEmojisUsedVariationModifier` array is currently *NOT* used, and this would be used only if we set the --useGithubUnicodes option on the command line when we execute
     `dart run tool/update_github_emojis.dart --useGithubUnicodes`

I include the ability to switch to githubs sequences as they are longer and include more information, but arrive at the same result.. (for example a family is not the family unicode emoji but a male adult, female adult and 2 children (male or female) unicodes all combined together with unicode variation selector and male/female sex selectors)...  They can end up a bit longer than the exising 'legacy' emoji sequences.
  ...but after playing with this a bunch it seems like maintaining the use of the legacy unicode sequences has the most advantages - all existing unit test using emojis still pass as the binary sequences are the same..

    The code ,as it is now, will use the github emoji unicode sequences for all/any new emojis that github currently has/or adds in the future.  This seems like the 'best of both worlds' as we get the new emojis, and we maintain binary compatibility with the old emoji set.